### PR TITLE
MG_IO_SIZE=256 on embedded platforms

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

**Review by scaprile — CHANGES_REQUESTED** [review]
- There are lots of changes for XMC, in mongoose.h where do these come from ?
- Please check the tests, there are buffer overflows when using Krypton
- 512 should be enough, lwIP uses 536 for TCP IIRC

File: mongoose.c [hunk 1]
```diff
@@ -10732,7 +10735,7 @@ void mg_tls_init(struct mg_connection *c, const struct mg_tls_opts *opts) {
       c->is_client ? MG_TLS_STATE_CLIENT_START : MG_TLS_STATE_SERVER_START;
 
   tls->skip_verification = opts->skip_verification;
-  tls->send.align = MG_IO_SIZE;
+  //tls->send.align = MG_IO_SIZE;
```
Comment (scaprile) [inline mongoose.c:10738]: Is this relevant ?
I mean, is it a desired/needed change or a debugging leftover ?

File: mongoose.c [hunk 2]
```diff
@@ -9847,7 +9847,9 @@ static void mg_tls_encrypt(struct mg_connection *c, const uint8_t *msg,
 #if CHACHA20
   (void) tag;  // tag is only used in aes gcm
   {
-    uint8_t *enc = (uint8_t *) malloc(8192);
+    size_t maxlen = MG_IO_SIZE > 16384 ? 16384 : MG_IO_SIZE;
+    uint8_t *enc = (uint8_t *) calloc(1, maxlen + 256 + 1);
+    //MG_INFO(("%lu %lu", msgsz, MG_IO_SIZE));
```
Comment (scaprile) [inline mongoose.c:9852]: leftover ?

File: mongoose.c [hunk 3]
```diff
@@ -9916,7 +9918,8 @@ static int mg_tls_recv_record(struct mg_connection *c) {
   nonce[11] ^= (uint8_t) ((seq) &255U);
 #if CHACHA20
   {
-    uint8_t *dec = (uint8_t *) malloc(msgsz);
+    //MG_INFO(("AAAAA %lu", msgsz));
```
Comment (scaprile) [inline mongoose.c:9921]: leftover

**Review by scaprile — COMMENTED** [review]
- Please confirm that .align removal
- Let me know if you want me to take overaddress those leftover comments removal
 
PS: There was a change in the driver order (why ?) that's where those XMC differences came from

**Review by scaprile — COMMENTED** [review]
- Please confirm that .align removal
- Let me know if you want me to take over and address those leftover comments removal

PS: There was a change in the driver order (why ?) that's where those XMC differences came from

File: mongoose.c (hunk 1 above)
Comment (cpq) [reply mongoose.c:10738]: send.align is set in net.c where all 3 buffers are initialised
https://github.com/cesanta/mongoose/blob/7923d1e930b661d699000643bc73715f6f0ebe30/src/net.c#L132

File: mongoose.c (hunk 2 above)
Comment (cpq) [reply mongoose.c:9852]: Yeah

File: mongoose.c (hunk 3 above)
Comment (cpq) [reply mongoose.c:9921]: It is !

**Comment by cpq** [comment]
Let's face it.. Merging is the only way to go..

**Review by scaprile — APPROVED** [review]

**Pull request merged**